src/server/cluster/cluster_slot_migration.cc

```diff
@@ -138,8 +142,23 @@ void ClusterSlotMigration::MainMigrationFb() {
                  << ec.message();
   }
 
-  if (IsFinalized()) {
+  bool success = true;
+  if (success = IsFinalized(); success) {
     state_ = MigrationState::C_FINISHED;
+
+    const auto added_slots = ToSlotSet(slots_);
+
+    // TODO it's a bug need to rewrite with mutex or fix tl_cluster_config because it's not thread
+    // local
+    cluster_family_->cluster_config()->AddSlots(added_slots);
```

Comment:
I think you should add the slots only if you receive OK on DFLYMIGRATE ACK

Reply:
why not use tl_cluster_config ?

Reply:
Add slots is not a problem even if don't get OK, because it's an error processing situation. But if we add slots only after ok we can lose requests from the clients because the source node removes slots but the target node doesn't add them so nobody can process requests